### ISSUE TITLE
[import/no-unresolved]: Do not report properly configured self resolution

### ISSUE THREAD

**Issue by coreyfarrell** [benmosher/eslint-plugin-import#1649]
In one of my modules I decided to use self resolution from testing as this helps to verify that my `exports` is setup and pointing to the correct file.  My package.json contains `"exports": "./index.js"` and `import lib from 'my-pkg-name';` works but `import/no-unresolved` reports this as an error.

For now I've added `// eslint-disable-next-line import/no-unresolved` but it would be nice to avoid this.

**Comment by ljharb** [comment]
Indeed; this is new as of node 13.7; we should support it (altho only in supported node versions).

**Comment by coreyfarrell** [comment]
One comment about only in supported node versions, it would be very useful to have an option to forcefully mark it as supported.  For any package which declares `package.json#exports` I always write a test file that uses self-reference import.  That self-reference test is conditionally run only in node.js 13.7.0+ to test the exports field even when `engines.node` is declared as `>=10`.

**Comment by ljharb** [comment]
I agree; we should probably have a setting for self resolution that defaults to "auto", and can be set to "forbid" or "allow".

**Comment by khan-opm** [comment]
How about `"import/no-unresolved": [2, { "ignore": ["^my-pkg-name(/(subexport1|subexport2))?$"] }]` to resolve the error?

Self-referencing is a new feature from 13.7.0, so I'd prefer to make sure my project requires node >= 13.7 rather than an option in .eslintrc.
